plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -2053,6 +2077,36 @@ static auto maybe_make_debug_time_logger() -> std::optional<decltype(make_debug_
    }
 }
 
+void block_only_sync::schedule() {
+   if (!_pending) {
+      // wait one second to see if we can actually get the block from net plugin before we try to resync from block vault
+      _start_sync_timer.expires_from_now(boost::posix_time::seconds(1));
+      _pending = true;
+      _start_sync_timer.async_wait(
+          app().get_priority_queue().wrap(priority::high, [this](const boost::system::error_code& ec) {
+             if (ec != boost::asio::error::operation_aborted) {
+                auto id = _impl->chain_plug->chain().last_irreversible_block_id();
+                fc_dlog(_log, "Attempt to resync from block vault");
+                _impl->blockvault->sync(&id, *this);
+             }
+             _pending = false;
+          }));
+   }
+}
+
+void block_only_sync::on_snapshot(const char*) {
+   EOS_THROW(producer_exception, "Attempting to resync from blockvault encountered a snapshot and the node must restart to continue!");
+}
+
+void block_only_sync::on_block(eosio::chain::signed_block_ptr block) {
+   try {
+      _impl->on_sync_block(block, true);
```

Comment:
we pass true here but I do not believe we are sure that is correct.  If this node has been initialized with a snapshot and has _no_ other input and constructs a block rejected by block-vault then we cannot verify connectivity.  On that edge case we would want to pass `false`.  

That case would only trigger if when we called `sync` the head_block was not available from controller.  in that case we would not verify connectivity for any block at exactly head_block_num + 1

Reply:
@huangminghuang this looks unaddressed still?

Reply:
That has been changed.